Plugins/SwiftLintPlugin/SwiftLintPlugin.swift

```diff
@@ -0,0 +1,33 @@
+import PackagePlugin
+
+@main
+struct SwiftLintPlugin: BuildToolPlugin {
+    func createBuildCommands(context: PackagePlugin.PluginContext, target: PackagePlugin.Target) async throws -> [PackagePlugin.Command] {
+        [
+            .buildCommand(
+                displayName: "SwiftLint",
+                executable: try context.tool(named: "swiftlint").path,
+                arguments: []
+            ),
+        ]
+    }
+}
+
+#if canImport(XcodeProjectPlugin)
+import XcodeProjectPlugin
+
+extension SwiftLintPlugin: XcodeBuildToolPlugin {
+
+    func createBuildCommands(context: XcodePluginContext, target: XcodeTarget) throws -> [Command] {
+        [
+            .buildCommand(
+                displayName: "SwiftLint",
+                executable: try context.tool(named: "swiftlint").path,
+                arguments: [
+                    "--cache-path", "\(context.pluginWorkDirectory)"
```

Comment:
Where does this resolve to? Somewhere semi-private where we can write our cache data to without bothering the user? I assume this is necessary for sandboxing reasons and we're not allowed to write to caches directory [here](https://github.com/realm/SwiftLint/blob/49ace1729f4aba727500d37a5223026dc5dfaf10/Source/SwiftLintFramework/Extensions/Configuration%2BCache.swift#L80)?

Reply:
Yes. This is a temporary directory inside `Derived Data`. 

From the Swift Package Manager documentation/proposal:

```swift
/// The path of a writable directory into which the plugin or the build
/// commands it constructs can write anything it wants. This could include
/// any generated source files that should be processed further, and it
/// could include any caches used by the build tool or the plugin itself.
/// The plugin is in complete control of what is written under this di-
/// rectory, and the contents are preserved between builds.
///
/// A plugin would usually create a separate subdirectory of this directory
/// for each command it creates, and the command would be configured to
/// write its outputs to that directory. The plugin may also create other
/// directories for cache files and other file system content that either
/// it or the command will need.
public let pluginWorkDirectory: PackagePlugin.Path
```

The [initial proposal](https://github.com/apple/swift-evolution/blob/main/proposals/0303-swiftpm-extensible-build-tools.md) also includes that explanation as well as hints to future improvements regarding user-defined options for plugins.